packages/core/src/framebuffer/FramebufferSystem.js

```diff
@@ -82,6 +86,44 @@ export class FramebufferSystem extends System
                 this.writeDepthTexture = false;
             }
         }
+        else
+        {
+            // WebGL2
+            // cache possible MSAA samples
+            this.msaaSamples = gl.getInternalformatParameter(gl.RENDERBUFFER, gl.RGBA8, gl.SAMPLES);
+        }
+    }
+
+    /**
+     * Detects number of samples that is not more than a param but as close to it as possible
+     *
+     * @param {number} samples number of samples
+     * @returns {number} recommended number of samples
+     */
+    detectSamples(samples)
```

Comment:
do we need this? Can we rather, just check values are PO2?

Reply:
I really don't know what to put there. This thing works and is covered by a test. I don't know which values are available on different systems, I didn't research it.

Reply:
I think we should just enforce the use of the consts? 

export enum MSAA_QUALITY {
    NONE = 0,
    LOW = 0,
    MEDIUM = 4,
    HIGH = 8
}

Reply:
![image](https://user-images.githubusercontent.com/695831/72533348-15c18a80-3886-11ea-9bbd-0f1cd6f53f96.png)

Reply:
it looks like there's no 16: https://webglstats.com/webgl2/parameter/SAMPLES

Maybe its my stupid time - I don't know what to write with those IF's. There will be more code in that case.